### PR TITLE
Auto release process

### PR DESCRIPTION
**Pull request by 2m**
This is my attempt at the automated release process. I literally got stuck while uploading docs from Travis. Rsync freezes promptly after starting up with:

```
sending incremental file list
rsync: connection unexpectedly closed (9 bytes received so far) [sender]
rsync error: error in rsync protocol data stream (code 12) at io.c(605) [sender=3.0.9]
```

That is blocking automating pushing docs to the server.

Also I had to created private/public keypair and encrypt the private part to `.travis_rsa_id.enc` which is then decrypted and used to upload docs to server during the CI build. I was not able to reuse the keypair managed by Travis for ssh, therefore this key/pair.

With this merged building and publishing JARs after creating GitHub release will work. However docs will have to be pushed manually.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
+    && chmod 600 $HOME/.ssh/travis_id_rsa
+
+deploy:
+  provider: script
+  # To restrict upload only to a certain directory prepend the key on the remote host with
+  # command="$HOME/bin/rrsync ~/<directory>/",no-agent-forwarding,no-port-forwarding,no-pty,no-user-rc,no-X11-forwarding ssh-rsa ...
+  script:
+    - sbt -J-XX:ReservedCodeCacheSize=128m +publish +unidoc docs/paradox
+      && rsync -rv -e "ssh -o StrictHostKeyChecking=no -l akkarepo -i $HOME/.ssh/travis_id_rsa" docs/target/paradox/site/ repo.akka.io:docs/$TRAVIS_TAG/
+      && rsync -rv -e "ssh -o StrictHostKeyChecking=no -l akkarepo -i $HOME/.ssh/travis_id_rsa" target/scala-2.12.0-RC2/unidoc/ repo.akka.io:api/$TRAVIS_TAG/
```
Comment (patriknw) [inline .travis.yml]: this path `scala-2.12.0-RC2` will break soon, will we notice that?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -85,6 +85,5 @@ Example:
 ## Releasing
 
 1. Create a [new release](https://github.com/akka/alpakka/releases/new) with the next tag version (e.g. `v0.3`), title and release decsription including notable changes mentioning external contributors.
-2. Fetch tags to your local clone: `git fetch --tags`.
-3. Publish signed release: `sbt publishSigned`.
-4. Stage and close [sonatype repository](https://oss.sonatype.org/).
+2. Travis CI will start a [CI build](https://travis-ci.org/2m/alpakka/builds) for the new tag and publish artifacts.
```
Comment (patriknw) [inline CONTRIBUTING.md]: https://travis-ci.org/akka/alpakka

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -85,6 +85,5 @@ Example:
...
+3. Sync artifacts of every module to Maven Central from [Bintray](https://bintray.com/akka/maven/). You must by part of the Akka team on Bintray.
```
Comment (patriknw) [inline CONTRIBUTING.md]: s/by/be/

File: .travis.yml (hunk 1 above)
Comment (johanandren) [inline .travis.yml]: 2.12 now :)

Comment (johanandren) [inline .travis.yml]: Can we steer unidoc to put it in a directory that is the same all the time?

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
+addSbtPlugin("com.dwijnand"                      % "sbt-dynver"       % "1.0.0+15-3655d014")
 addSbtPlugin("com.lightbend.paradox"             % "sbt-paradox"      % "0.2.5")
 addSbtPlugin("com.eed3si9n"                      % "sbt-unidoc"       % "0.3.3")
-addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings" % "0.2.4")
+
+// Reenable after https://github.com/ThoughtWorksInc/sbt-api-mappings/issues/5 is fixed and released
+//addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings" % "0.2.4")
+
+addSbtPlugin("me.lessis" % "bintray-sbt" % "0.3.0-8-g6d0c3f8")
+resolvers += Resolver.url("2m-sbt-plugin-releases", url("https://dl.bintray.com/2m/sbt-plugin-releases/"))(Resolver.ivyStylePatterns)
```
Comment (johanandren) [inline project/plugins.sbt]: Why do we need a 2m specific resolver here?

Comment (ktoso) [inline project/plugins.sbt]: So he can inject his hacked artifacts! 😜 
Probably an omission? If not then should be documented

File: .travis.yml (hunk 1 above)
Comment (2m) [inline .travis.yml]: Yes. We should make it more robust here.

**Comment by 2m** [comment]
We should make it sync to maven central automatically before it gets merged: https://github.com/softprops/bintray-sbt/blob/0.3.0/src/main/scala/BintrayKeys.scala#L62-L63

File: project/plugins.sbt (hunk 4 above)
Comment (johanandren) [inline project/plugins.sbt]: Ok, cool! So when it get's into bintray-sbt official we don't have to use custom resolver here?

**Comment by patriknw** [comment]
LGTM

**Comment by 2m** [comment]
Tried to release an RC with these changes and got this:
```
[error] (amqp/*:bintraySyncMavenCentral) failed to sync akka/akka-stream-alpakka-amqp@0.1-RC1 with maven central: {"message":"In order to sync to Maven Central your package must be included in the JCenter repository"}
```

Need to wait until this package is included to JCenter. This is one time thing. Other packages where included already and are normally included within couple of hours since request.

**Pull request merged**